language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -37,6 +37,48 @@ import DA.Daml.Project.Consts
 import DA.Daml.Project.Types
 import qualified DA.Daml.Project.Types as DATypes
 
+-- Referenced from 'writePackageJson'. Lifted here for easy
+-- maintenance.
+data ConfigConsts = ConfigConsts
+  { pkgDependencies :: HMS.HashMap NpmPackageName NpmPackageVersion
+  , pkgDevDependencies :: HMS.HashMap NpmPackageName NpmPackageVersion
+  , pkgScripts :: HMS.HashMap ScriptAlias Script
+  }
+configConsts :: T.Text -> ConfigConsts
+configConsts sdkVersion = ConfigConsts
+  { pkgDependencies = HMS.fromList
+      [ (NpmPackageName "@mojotech/json-type-validation", NpmPackageVersion "^3.1.0")
+      , (NpmPackageName "@daml/types", NpmPackageVersion sdkVersion)
+      ]
+  , pkgDevDependencies = HMS.fromList
+      [ (NpmPackageName "@typescript-eslint/eslint-plugin", NpmPackageVersion "2.11.0")
+      , (NpmPackageName "@typescript-eslint/parser", NpmPackageVersion "2.11.0")
+      , (NpmPackageName "eslint", NpmPackageVersion "^6.7.2")
+      , (NpmPackageName "typescript", NpmPackageVersion "~3.7.3")
+      ]
+  , pkgScripts = HMS.fromList
+      [ (ScriptAlias "build", Script "tsc --build")
+      , (ScriptAlias "lint", Script "eslint --ext .ts src/ --max-warnings 0")
+      ]
+  }
+newtype NpmPackageName = NpmPackageName {unNpmPackageName :: T.Text} deriving (Eq, Show, Hashable)
+instance FromJSON NpmPackageName where parseJSON = fmap NpmPackageName . parseJSON
+instance ToJSON NpmPackageName where toJSON = toJSON . unNpmPackageName
+instance ToJSONKey NpmPackageName where toJSONKey = ToJSONKeyText unNpmPackageName (text . unNpmPackageName)
+
+newtype NpmPackageVersion = NpmPackageVersion {unNpmPackageVersion :: T.Text} deriving (Eq, Show, Hashable)
+instance FromJSON NpmPackageVersion where parseJSON = fmap NpmPackageVersion . parseJSON
+instance ToJSON NpmPackageVersion where toJSON = toJSON . unNpmPackageVersion
+
+newtype ScriptAlias = ScriptAlias {unScriptAlias :: T.Text} deriving (Eq, Show, Hashable)
```

Comment:
Alias sounds like it already has a name whereas you are defining the name here. So maybe `ScriptName` would be a better name or `ScriptId`.

Reply:
Ok